pkg/component/manager.go

```diff
@@ -24,37 +24,38 @@ import (
 	"github.com/spf13/afero"
 )
 
+// ResolvePath resolves a given path to a module and a component.
 func ResolvePath(ksApp app.App, path string) (Module, Component, error) {
 	isDir, err := isComponentDir2(ksApp, path)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "check for namespace directory")
+		return nil, nil, errors.Wrap(err, "checking for module directory")
 	}
 
 	if isDir {
-		ns, err := GetModule(ksApp, path)
+		m, err := GetModule(ksApp, path)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, errors.Wrapf(err, "retrieving module %s", path)
 		}
 
-		return ns, nil, nil
+		return m, nil, nil
 	}
 
-	module, cName, err := checkComponent(ksApp, path)
+	moduleName, componentName, err := extractPathParts(ksApp, path)
```

Comment:
we're passing `path` in as `name` to extractPathParts ... if we've gotten here then `path` is not a path. previous to here `path` could actually be a path (if isDir is true it's an actual path right?) ... i feel like `name` feels better all the way up on this where `name` could be a path and you early out if it is.

Reply:
It's a path and not a name. I updated the function.